osrparse/replay.py

```diff
@@ -79,13 +79,22 @@ def parse_replay_data(replay_data_str, mode):
 
         rng_seed = None
         play_data = []
-        for event in events:
+        for i, event in enumerate(events):
             time_delta = int(event[0])
             x = event[1]
             y = event[2]
             keys = int(event[3])
 
-            if time_delta == -12345 and event == events[-1]:
+            # I don't really know why these frames exist, but lazer removes them
+            # and they can cause issues for minigame replays - e.g., mania
+            # interprets x as keys.
+            # See
+            # https://github.com/ppy/osu/blob/6a04708a7e9801949c6c7ac7ddf6a4d7
+            # fa0835e5/osu.Game/Scoring/Legacy/LegacyScoreDecoder.cs#L290-L294.
+            if i < 2 and float(x) == 256 and float(y) == -500:
+                continue
+
+            if time_delta == -12345 and i == len(events) - 1:
                 rng_seed = keys
```

Comment:
I'm a bit confused - if you mean "does the `rng_seed = keys` branch get hit when `i < 2 and float(x) == 256 and float(y) == -500` is true", then the answer is no, because `continue` skips the rest of the loop body.

If you mean "does the `rng_seed = keys` branch get hit if there are only two frames in the replay and both satisfy (x, y) == (256, -500)", then the answer is also no. Strictly speaking, this doesn't actually match lazer behavior - lazer checks `t == -12345` before skipping any frames. I guess we should probably change that for absolute correctness, though it may have been by happenstance and not deliberate choice that lazer parses this way.

Reply:
As discussed, have gone ahead and made the change to match lazer. Because we might as well.